src/shared/modules/dbMeta/dbMetaDuck.js

```diff
@@ -312,162 +312,166 @@ export const dbMetaEpic = (some$, store) =>
-            const state = store.getState()
-            const db = getUseDb(state)
-            if (db === 'system') {
-              store.dispatch(updateMeta([]))
-              return Rx.Observable.of(null)
-            }
-
-            return Rx.Observable.fromPromise(
-              bolt.routedReadTransaction(
-                metaQuery,
-                {},
-                {
-                  useCypherThread: shouldUseCypherThread(store.getState()),
-                  onLostConnection: onLostConnection(store.dispatch),
-                  ...getBackgroundTxMetadata({
-                    hasServerSupport: canSendTxMetadata(store.getState())
-                  })
-                }
-              )
-            ).catch(e => {
-              store.dispatch(updateMeta([]))
-              return Rx.Observable.of(null)
-            })
-          })
-          .do(res => {
-            if (res) {
-              store.dispatch(updateMeta(res))
-            }
-          })
-          // Server configuration
           .mergeMap(() =>
-            Rx.Observable.fromPromise(
-              bolt.directTransaction(
-                'CALL dbms.listConfig()',
-                {},
-                {
-                  useCypherThread: shouldUseCypherThread(store.getState()),
-                  ...getBackgroundTxMetadata({
-                    hasServerSupport: canSendTxMetadata(store.getState())
-                  })
-                }
-              )
-            )
-              .catch(e => {
-                return Rx.Observable.of(null)
-              })
-              .do(res => {
-                if (!res) return Rx.Observable.of(null)
-                const settings = res.records.reduce((all, record) => {
-                  const name = record.get('name')
-                  let value = record.get('value')
-                  if (name === 'browser.retain_connection_credentials') {
-                    let retainCredentials = true
-                    // Check if we should wipe user creds from localstorage
-                    if (
-                      typeof value !== 'undefined' &&
-                      isConfigValFalsy(value)
-                    ) {
-                      retainCredentials = false
-                    }
-                    store.dispatch(setRetainCredentials(retainCredentials))
-                  } else if (name === 'browser.allow_outgoing_connections') {
-                    // Use isConfigValFalsy to cast undefined to true
-                    value = !isConfigValFalsy(value)
-                  } else if (name === 'dbms.security.auth_enabled') {
-                    let authEnabled = true
-                    if (
-                      typeof value !== 'undefined' &&
-                      isConfigValFalsy(value)
-                    ) {
-                      authEnabled = false
-                    }
-                    store.dispatch(setAuthEnabled(authEnabled))
+            Rx.Observable.forkJoin([
```

Comment:
Not your fault, but these observables are crazy hard to read. Do you think it can be refactored to smaller pieces and glued together here?
What's super important is that this doesn't stop listening for new events after errors has happened.

Reply:
I agree, I thought a bit about how to break apart the chain and the options I could think off are either to refactor the chain to return actions instead of dispatching them, so parts can be broken out into pure functions, or to replace the large chain with several small chains, each with their own actions. I didn't do it in this branch as either of those options would be a bigger change and hard to review mixed with these changes. Maybe we should consider the big size of the epic tech debt and add a card in our backlog to break it apart?

Reply:
Added tech debt card to backlog